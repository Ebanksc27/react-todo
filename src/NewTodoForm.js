import React, { useState } from 'react';

function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState('');

  const uniqueId = () => {
    return Date.now().toString(36) + Math.random().toString(36).substr(2);
  };

  const handleSubmit = e => {
    e.preventDefault();
    addTodo({ task, id: uniqueId() }); 
    setTask('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text" 
        value={task} 
        onChange={e => setTask(e.target.value)} 
        placeholder="New task" 
      />
      <button type="submit">Add Todo</button>
    </form>
  );
}

export default NewTodoForm;
