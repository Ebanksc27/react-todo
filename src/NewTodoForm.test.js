import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders NewTodoForm component without crashing', () => {
  // Pass a mock addTodo function as a prop
  const addTodo = jest.fn();
  render(<NewTodoForm addTodo={addTodo} />);
});

test('NewTodoForm component snapshot', () => {
  const addTodo = jest.fn();
  const { asFragment } = render(<NewTodoForm addTodo={addTodo} />);
  expect(asFragment()).toMatchSnapshot();
});
