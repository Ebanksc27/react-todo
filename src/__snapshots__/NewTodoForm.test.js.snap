// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm component snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="New task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
